### ISSUE TITLE
adding a bundle, removing the bundle, and then adding the same bundle behaves strangely

### ISSUE THREAD

**Issue by matthewrsj** [clearlinux/swupd-client#254]
There appears to be some staging directory strangeness when adding a bundle:
```
swupd bundle-add rust-basic
```
then removing that bundle
```
swupd bundle-remove rust-basic
```
then adding that bundle back again
```
swupd bundle-add rust-basic
swupd-client bundle adder 3.12.1
   Copyright (C) 2012-2017 Intel Corporation

Attempting to download version string to memory
Downloading packs...
	...100%
Installing bundle(s) files...
	...0%Path /usr/bin/cargo is missing on the file system
Path /usr/bin/rust-gdb is missing on the file system
Path /usr/bin/rust-lldb is missing on the file system
Path /usr/bin/rustc is missing on the file system
Path /usr/bin/rustdoc is missing on the file system
	...35%Path /usr/lib64/libarena-5fe3094a53c2fc91.so is missing on the file system
Path /usr/lib64/libfmt_macros-091a58a94fe20486.so is missing on the file system
...
<snip>
...
Path /usr/share/man/man1/cargo.1 is missing on the file system
	...42%Path /usr/share/man/man1/rustc.1 is missing on the file system
Path /usr/share/man/man1/rustdoc.1 is missing on the file system
	...43%Path /usr/share/zsh/site-functions/_cargo is missing on the file system
	...87%
Calling post-update helper scripts.
Bundle(s) installation done.
```

In the case above the bundle at least seems to be fully installed to the system.

Wiping the state dir in between the bundle adds fixes the issue of course.

**Comment by matthewrsj** [comment]
The messages indicate bundle-add is falling back to the verify_fix_path, even though the bundle files should be present in the staging directory.

**Comment by phmccarty** [comment]
@matthewrsj do_staging() can fail for a number of reasons. It's unclear to me which one of the errors is being hit.

**Comment by phmccarty** [comment]
I am also seeing these messages, but then repeating another bundle-remove/bundle-add makes them go away.

**Comment by matthewrsj** [comment]
bundle-add is changing the permissions of the `staged` directory somehow to `750` when it drops into the `verify_fix_path`, causing `ensure_root_owned_dir` to remove everything under `staged/` when it recreates it with the right permissions (on the next run).

**Comment by matthewrsj** [comment]
This is actually a swupd-server issue, creating `staging` and `delta` with `750` permissions

**Comment by phmccarty** [comment]
I verified that running `bundle-add` starting from an empty state directory resulted in `/var/lib/swupd/staged` having `750` permissions. If I ran `chmod 700 /var/lib/swupd/staged` prior to running the `bundle-remove` command, the `staged/` directory does _not_ get wiped and the subsequent `bundle-add` succeeds.

So, making `swupd_make_pack` create delta and staged with proper permissions is definitely the right fix.
